pkg/ebpf/c/tracee.bpf.c

```diff
@@ -5051,18 +5053,86 @@ static __always_inline bool skb_revalidate_data(struct __sk_buff *skb, uint8_t *
     return true;
 }
 
+static __always_inline int is_http_protocol(struct __sk_buff *skb, uint8_t *head, uint8_t *tail, u32 l7_off) {
+#define http_min_len 7  // longest http command is "DELETE "
+
+    char *http = (void *)head + l7_off;
+
+    char http_str[http_min_len];
+    __builtin_memset((void *)&http_str, 0, sizeof(char)*http_min_len);
+
+    // get first http_min_len of layer 7
+    if ( tail-head < (u64)skb->len) {
+        // non-linear skb.
+        // use helper to load data into http_str.
+        if (bpf_skb_load_bytes(skb, l7_off, http_str, http_min_len) < 0){
+            // failed loading data into http_str - return.
+            return 0;
+        }
+    }
+    else if ((void *)http + sizeof(char)*http_min_len < (void *)tail) {
+        #ifdef CORE
+            bpf_core_read((void *)&http_str, sizeof(char)*http_min_len, http);
+        #else
+            bpf_probe_read((void *)&http_str, sizeof(char)*http_min_len, http);
+        #endif
```

Comment:
Why not using READ_KERN instead?

Reply:
because i need to state the size - how many bytes to copy (in this case 7).